### ISSUE TITLE
Weighting Module Picked the Wrong Corsika Primary (Trac #1376)

### ISSUE THREAD

**Issue by icecube-trac** [icecube-trac/tix2#1368]
When weighting module get_weighted_primary is used to find MCPrimary on dataset 10668 and 10809 corsika simulation, it picked primaries that are not one of the 5 components. It's very likely caused by the fact that those datasets are PDG-coded simulation while the weighting module is written for CORSIKA-style particle type codes. The module may need to be updated 

A file contains wrongly picked corsika primaries can be found here:/data/user/yxu/IC86/private/hdf5/l3_IC86_cors_1std_SpiceLea.i3.bz2

Migrated from https://code.icecube.wisc.edu/ticket/1376
```json
{
    "status": "closed", 
    "changetime": "2015-11-19T18:26:01", 
    "description": "When weighting module get_weighted_primary is used to find MCPrimary on dataset 10668 and 10809 corsika simulation, it picked primaries that are not one of the 5 components. It's very likely caused by the fact that those datasets are PDG-coded simulation while the weighting module is written for CORSIKA-style particle type codes. The module may need to be updated \n\nA file contains wrongly picked corsika primaries can be found here:/data/user/yxu/IC86/private/hdf5/l3_IC86_cors_1std_SpiceLea.i3.bz2", 
    "reporter": "yiqian.xu", 
    "cc": "", 
    "resolution": "fixed", 
    "_ts": "1447957561144991", 
    "component": "combo simulation", 
    "summary": "Weighting Module Picked the Wrong Corsika Primary", 
    "priority": "normal", 
    "keywords": "", 
    "time": "2015-10-01T16:17:07", 
    "milestone": "", 
    "owner": "jvansanten", 
    "type": "defect"
}
```

**Comment by icecube-trac** [comment]
Trac update at 20151001T16:20:21: 
* david.schultz changed reporter from "icecube" to "yiqian.xu"
* david.schultz changed component from "tools/ports" to "simulation"
* david.schultz changed owner from "" to "jvansanten"
* david.schultz changed status from "new" to "assigned"

**Comment by icecube-trac** [comment]
Trac update at 20151119T18:26:01: 
* jvansanten changed resolution from "" to "fixed"
* jvansanten changed status from "assigned" to "closed"
* jvansanten commented:

> In changeset 139686:
> 
> ```text
> #!CommitTicketReference repository="IceCube" revision="139686"
> Identify weighted primary by energy if we have it. Fixes #1368.
> ```
